lib/Service/ApiService.php

```diff
@@ -67,4 +81,35 @@ public function sendFile(int $fileId): void {
 			],
 		);
 	}
+
+	public function searchMessages(string $userId, string $term, int $offset = 0, int $limit = 10): array {
+		$arguments = [
+			'format' => 'json',
+			'query' => '*' . $term . '*' ,
+		];
+
+		$paperlessURL = rtrim($this->config->url, '/') . '/api/documents/?' . http_build_query($arguments);
+		$result = $this->client->get($paperlessURL,
+			[
+				'headers' => array_merge(
+					$this->getAuthorizationHeaders(),
+					[
+						'Content-Type' => 'application/json',
+						'Accept' => 'application/json'
+					]
+				)
+			]
+		);
+
+		$body = $result->getBody();
+		$json_body = json_decode($body, true);
+
+		if (isset($json_body['error'])) {
+			return (array)$json_body;
+		}
```

Comment:
I don't see anything like this in the API docs. Please check in the source code of paperless-ngx how an error response will look like and adjust it here.

Reply:
I don't find any explicit error key in the API responses. However, I noticed that the API returns an HTML response when the URL is incorrect.

So, I’m checking for an HTML key in the JSON. Let me know if you have different ideas.